### PR TITLE
fixes #1685 - Windows DNS: Secure connection using GSS-TSIG

### PR DESCRIPTION
**Pull request by ohadlevy**
Signed-off-by: theforemanuser123 oliver_weinmann@gmx.de

### CODE REVIEW

**Comment by ohadlevy** [comment]
it looks good, however, I would say that the kerb stuff probably need to exists in a class by itself, more minor comments bellow

File: lib/proxy/dns/bind.rb [hunk 1]
```diff
@@ -60,11 +80,60 @@ def find_nsupdate
...
+    def find_kinit
+      @kinit = which("kinit", "/usr/bin")
+      unless File.exists?("#{@kinit}")
+        logger.warn "unable to find kinit binary, maybe missing krb5-user package?"
+        raise "unable to find kinit binary"
+      end
+    end
+
+    def find_krb5tgt
+      krb5tgt = system("klist | grep -i #{SETTINGS.dns_tsig_principal} 2>&1")
```
Comment (ohadlevy) [inline lib/proxy/dns/bind.rb:92]: what if klist is not in path? I guess we need to do which(klist..).

also, if we move this code into its own line, we could parse it without reverting to grep :)

File: lib/proxy/dns/bind.rb [hunk 2]
```diff
@@ -60,11 +80,60 @@ def find_nsupdate
...
+      if krb5tgt == false
+        exp = 1
+        now = 0
+        @loggermsg = "unable to find kerberos ticket. Trying to aquire a valid TGT..."
+        @raisemsg = "unable to find kerberos ticket. Trying to aquire a valid TGT..."
```
Comment (ohadlevy) [inline lib/proxy/dns/bind.rb:97]: these error messages are never used i think

File: lib/proxy/dns/bind.rb [hunk 3]
```diff
@@ -60,11 +80,60 @@ def find_nsupdate
...
+        kinit
+      else
+        format = "%d/%m/%y %H:%M"
+        krbexp = `klist | grep A.SPACE.CORP | grep '/' | awk -F ' ' '{print $3,$4}'`
```
Comment (ohadlevy) [inline lib/proxy/dns/bind.rb:101]: I guess the kerb domain needs to be extracted into a setting value?

File: lib/proxy/dns/bind.rb (hunk 1 above)
Comment (r0k5t4r) [inline lib/proxy/dns/bind.rb:92]: what if klist is not in path? I guess we need to do which(klist..).

--> I added which klist

also, if we move this code into its own line, we could parse it without reverting to grep :)

hmm can you give me a small hint, how to do this. Basically all I did was copy & paste for most of the code. :)

File: lib/proxy/dns/bind.rb (hunk 2 above)
Comment (r0k5t4r) [inline lib/proxy/dns/bind.rb:97]: fixed. I forgot the "_" in the var names.

File: lib/proxy/dns/bind.rb (hunk 3 above)
Comment (r0k5t4r) [inline lib/proxy/dns/bind.rb:101]: Forgot to remove it. :( I have added a new setting value "dns_tsig_realm"

**Comment by r0k5t4r** [comment]
t looks good, however, I would say that the kerb stuff probably need to exists in a class by itself, more minor comments bellow

--> Would this be as easy as creating a new file e.g. kerberos.rb, put the code in there and put a require kerberos.rb in bind.rb?

**Comment by r0k5t4r** [comment]
closing this one as it's a duplicate of #29.

**Comment by domcleal** [comment]
Please ignore the test failure comment, the job was misconfigured, I believe it passes.  Deleting the bot's comments should cause it to get retested.

**Comment by domcleal** [comment]
#77 has been merged for Foreman 1.2 to do the same via rkerberos instead of klist/kinit.  Thanks for the contribution!

**Pull request closed without merging**